src/components/implementation/no_interface/cFE_booter/ostask.c

```diff
@@ -0,0 +1,793 @@
+#include <cos_kernel_api.h>
+#include <sl.h>
+#include <sl_lock.h>
+
+#include "cFE_util.h"
+#include "ostask.h"
+
+#include "gen/osapi.h"
+#include "gen/common_types.h"
+#include "gen/cfe_time.h"
+
+#define HZ_PAUSE (1000 * 1000)
+
+void
+timer_fn_1hz(void *d)
+{
+	cycles_t now, start;
+
+	rdtscll(start);
+	start += sl_usec2cyc(HZ_PAUSE);
+
+	while (1) {
+		rdtscll(now);
+		if (now > start) {
+			CFE_TIME_Local1HzISR(); /* input param is signum. but CFE_PSP_TimerHandler doesn't seem to use
+			                           it. */
+		}
+
+		sl_thd_block_periodic(0);
```

Comment:
just to make sure we aren't spinning in this loop for some strange interaction in the system, it might be smart to count how many times we've iterated without triggering the `now > start` condition, and assert if it goes above some level (say 64).